cse/WeightCalculation.py

```diff
@@ -49,8 +58,32 @@ def euclDistance(docWeights, queryWeights):
     return float(score)
 
 
-if __name__ == "__main__":
-    w1 = [1,4,0,1.4,23.0]
-    w2 = [76,5.7,1,13.5,3.5]
-
-    print(cosineSimilarity(w1, w2))
+class NormCache(object):
+    """
+    Singleton Cache from http://python-3-patterns-idioms-test.readthedocs.io/en/latest/Singleton.html.
+    Instance Holder
+    """
+    class __NormCache:
```

Comment:
FYI: The inner class is not necessary. There are ways to do it with one class.

Reply:
Then tell me...

Reply:
Im not sure which way is better.

```
class Singleton(object):
    """Basic chain class."""

    __instance = None

    def __new__(cls):

        """Create a singleton instance"""
        if cls.__instance is None:
            cls.__instance = object.__new__(cls)
        return cls.__instance

    def __init__(self):
        pass
```

Reply:
but in this case you just create a new `object` instance without any methods, aren't you?
```python
cls.__instance = object.__new__(cls)
```

Reply:
how would one define new attributes, methods, etc. of the `__instance`?